python/lsst/meas/algorithms/subtractBackground.py

```diff
@@ -0,0 +1,260 @@
+    )
+    weighting = pexConfig.Field(
+        doc="Use inverse variance weighting in calculation (valid only with useApprox=True)",
+        dtype=bool, default=True,
+    )
+
+    def validate(self):
+        pexConfig.Config.validate(self)
+        # Allow None to be used as an equivalent for "NONE", even though C++ expects the latter.
+        if self.algorithm is None:
+            self.algorithm = "NONE"
+
+class SubtractBackgroundTask(pipeBase.Task):
+    ConfigClass = SubtractBackgroundConfig
+    _DefaultName = "subtractBackground"
+
+    def run(self, exposure, background=None, stats=True, statsKeys=None):
+        """!Fit and subtract the background of an exposure
+
+        @param[in,out] exposure  exposure whose background is to be subtracted
+        @param[in,out] background  initial background model already subtracted from exposure
+            (an lsst.afw.math.BackgroundList). May be None if no background has been subtracted.
+        @param[in] stats  if True then measure the mean and variance of the full background model
+                        and record the results in the exposure's metadata
+        @param[in] statsKeys  key names used to store the mean and variance of the background
+            in the exposure's metadata (a pair of strings); if None then use ("BGMEAN", "BGVAR");
+            ignored if stats is false
+
+        @return an lsst.pipe.base.Struct containing:
+        - background  full background model (initial model with changes), an lsst.afw.math.BackgroundList
+
+        @throw RuntimeError if fitBackground returns a background of None
+        """
+        displayBackground = lsstDebug.Info(__name__).displayBackground
+
+        if background is None:
+            background = afwMath.BackgroundList()
+
+        maskedImage = exposure.getMaskedImage()
+
+        fitBg = self.fitBackground(maskedImage)
+        if not fitBg:
+            raise RuntimeError("Unable to estimate background for exposure")
+
+        fitBgImg = fitBg.getImageF()
+        maskedImage -= fitBgImg
+        background.append(fitBg)
+        del fitBgImg
+
+        if displayBackground > 1 or stats:
+            netBgImg = background.getImage()
+
+        if displayBackground > 1:
+            ds9.mtv(netBgImg, title="background", frame=3)
+
+        # Record statistics of the background in the bgsub exposure metadata.
+        # (lsst.daf.base.PropertySet)
+        if stats:
+            if statsKeys is None:
+                statsKeys = ("BGMEAN", "BGVAR")
+            mnkey, varkey = statsKeys
+            meta = exposure.getMetadata()
+            s = afwMath.makeStatistics(netBgImg, afwMath.MEAN | afwMath.VARIANCE)
+            bgmean = s.getValue(afwMath.MEAN)
+            bgvar = s.getValue(afwMath.VARIANCE)
+            meta.addDouble(mnkey, bgmean)
+            meta.addDouble(varkey, bgvar)
+
+        if displayBackground:
+            ds9.mtv(exposure, title="subtracted")
+
+        return pipeBase.Struct(
```

Comment:
Why return a `Struct` with a single element?  I know that it adds some future-proofing against changes where you might decide to return more information.  But I'm not sure you want that future-proofing, but instead you want things to fail because it's changed.